### ISSUE TITLE
When to fetch data for my screens?

### ISSUE THREAD

**Issue by mbrochh** [wix/react-native-navigation#365]
Sorry if this is a very stupid question. So far I have been building my apps with react-router and the Navigator component. In that scenario I have an App.js component that instantiates the `<Navigator>` and in that `<App>` component I can use the usual react lifecycle methods and do application-wide things (like checking if the user is logged-in, getting new code-push bundles and so on).

With react-native-nagvigation, I don't seem to have any parent component where I can do application-wide stuff. It seems like all my views are isolated entities that pushed or popped by the navigation. Furthermore, it seems that all main navigation components get instantiated at once and thus their componentDidMount gets executed immediately, even if they are not visible on screen right now.

In my old apps, I could always rely on the fact that anything that is not on screen, is unmounted, so once stuff gets visible, I can fetch my data from my APIs in componentDidMount. I can't do this with react-native-navigation any more.

So basically my question is: When and where should I fetch my application data? How do I know when a component will be mounted and unmounted?

**Comment by guyca** [comment]
Good question @mbrochh 
We also have an `App.js` class where we instantiate relevant screens, and since the static code there is executed once when the bundle is loaded, you can certainly also fetch data there since that's where your store is created and each Screen is connected to it.

The reason why all BottomTab screens are immediately created and mounted is to improve the responsiveness  of your app. This way the user doesn't see progress bars and loaders each time he navigates to a tab.

You should not use `componentDidMount` as an indication to a Screens visibility. We will expose proper Screen lifecycle methods as part of the [v2.0.0 milestone ](https://github.com/wix/react-native-navigation/milestone/1)

What works for us, is dispatching an action when the Application is created determining the appropriate course of action there (e.g Check if user is logged in, code push updates etc.)

```js
export default class App {
  constructor() {
    store.subscribe(this.onStoreUpdate.bind(this));
    store.dispatch(appActions.appInitialized());
  }

  onStoreUpdate() {
    const {root} = store.getState().app;

    // Handle new root
    if (this.currentRoot != root) {
    }
  }
}
```

Hope this answers your question
